solr/contrib/blob-directory/src/java/org/apache/solr/blob/BlobDirectory.java

```diff
@@ -61,20 +62,33 @@
   private final Object lock = new Object();
   private volatile boolean isOpen;
 
-  public BlobDirectory(Directory delegate, String blobDirPath, BlobStoreConnection blobStoreConnection) throws IOException {
+  public BlobDirectory(Directory delegate, String blobDirPath, BlobRepository blobRepository) throws IOException {
     super(delegate);
     this.blobDirPath = blobDirPath;
-    this.blobStoreConnection = blobStoreConnection;
+    this.blobRepository = blobRepository;
     pullMissingFilesFromRepo();
   }
 
   private void pullMissingFilesFromRepo() throws IOException {
     Set<String> localFileNames = new HashSet<>(Arrays.asList(in.listAll()));
-    blobStoreConnection.pull(blobDirPath, this::openOutput, blobFile -> {
-      //TODO: We could also check the size and checksum.
-      return !localFileNames.remove(blobFile.fileName());
+    MutableInt numPulledFiles = log.isInfoEnabled() ? new MutableInt() : null;
```

Comment:
Is the only reason you want to use MutableInt vs a primitive integer because MutableInt can be null?  If so, I think it's equally clear to have the log.isInfoEnabled check later and simply increment the numPulledFiles always.  I'd say _clearer_ because then the log.info call would be guarded by a clear "log.isInfoEnabled" that Solr's source code checks demand to see and which many people come to expect any way.

Reply:
I use a MutableInt to increment it inside the lambda expression for the file filter. I'll use log.isInfoEnabled() instead for the log, and I'll simplify to always create the MutableInt counter even if eventually it is not used because log info is not enabled.